packages/ember-principled-forms/tests/dummy/app/components/form-field/component.ts

```diff
@@ -6,42 +6,45 @@ import update from 'ember-object-update';
 import { Maybe } from 'true-myth';
 
 import Field, { validate } from 'principled-forms/field';
-import Number from 'principled-forms/field/number';
+import EmailField from 'principled-forms/field/email';
+import NumberField from 'principled-forms/field/number';
 import Form, { FromModel } from 'principled-forms/form';
+import Validity from 'principled-forms/validity';
 
 // @ts-ignore: Ignore import of compiled template
 import layout from './template';
 
 type User = {
   age: number;
-  name: Maybe<string>;
+  name: string;
+  email: Maybe<string>;
 };
 
 const userForm: FromModel<User> = user => ({
-  age: Number.required({ eager: false }),
-  name: Field.optional()
+  // DISCUSS: Does it actually every make sense for `eager` to be false?
```

Comment:
Heh, no: in fact it was talking with *you* about emails in particular that made me realize we needed it! We let this tell us whether `validate` should return `Unvalidated` or `Invalid` in the scenario where we're e.g. building up an email. `chris@krycho` shouldn't be validated as `Invalid` but `Unvalidated` when I type it in for the first time.  But once it has become `Validated` once (i.e. it starts in `Invalid` or `Valid`), then it always remains in one of the `Validated` states.

The `eager` flag is what tells us to do that behavior in the email-specific way.

Reply:
I think what I'm getting at here is this: Wouldn't we want, in all cases, for `validate` to return `Invalid` only after the field has been `Valid` at least once? Don't we always want the eager behavior? I can't think of a situation where we wouldn't.